src/cromshell/execution_status_count/command.py

```diff
@@ -0,0 +1,251 @@
+    :param workflow_metadata: Metadata of the workflow to process
+    :param do_expand_sub_workflows: Boolean, whether or not to print subworkflows
+    :return:
+    """
+    workflow_status = workflow_metadata.get("status")
+    print(
+        colored(
+            workflow_id + "\t" + workflow_status,
+            color=io_utils.TextStatusesColor.COLOR_UNDERLINED["color"],
+            attrs=io_utils.TextStatusesColor.COLOR_UNDERLINED["attrs"],
+        )
+    )
+    print_workflow_status(
+        workflow_metadata=workflow_metadata,
+        indent="\t",
+        expand_sub_workflows=do_expand_sub_workflows,
+    )
+
+
+def print_workflow_status(
+        workflow_metadata: dict, indent: str, expand_sub_workflows: bool
+) -> None:
+    """
+    Recursively runs through each task of a workflow metadata and calls function to
+    print task status counts
+    :param workflow_metadata: Metadata of the workflow to process
+    :param indent: Indent string given as "\t", used to indent print out
+    :param expand_sub_workflows:  Boolean, whether or not to print subworkflows
+    :return:
+    """
+    tasks = list(workflow_metadata["calls"].keys())
+
+    for task in tasks:  # For each task in given metadata
+        # If task has a key called 'subworkflowMetadata' in its first (zero) element
+        # (shard) and expand_sub_workflow parameter is set to true then rerun this
+        # function on that subworkflow
+        if (
+                "subWorkflowMetadata" in workflow_metadata["calls"][task][0]
+                and expand_sub_workflows
+        ):
+            sub_workflow_name = task
+            task_shards = workflow_metadata["calls"][sub_workflow_name]
+            print(f"{indent}SubWorkflow {sub_workflow_name}")
+
+            # For each element in total number of subworkflow calls get the subworkflow
+            # metadata. This loop will go through each shard if task is scattered
+            for i in range(len(task_shards) - 1):
+                sub_workflow_metadata = task_shards[i]["subWorkflowMetadata"]
+
+                print_workflow_status(
+                    workflow_metadata=sub_workflow_metadata,
+                    indent=indent + "\t",
+                    expand_sub_workflows=expand_sub_workflows,
+                )
+
+        # If no subworkflow is found then print status summary for task
+        else:
+            print_task_status(task, indent, workflow_metadata)
+
+
+def print_task_status(task: str, indent: str, workflow_metadata: dict) -> None:
+    """
+    Prints the task name and status count
+
+    :param task: Name of the task
+    :param indent: Indent string given as "\t", used to indent print out
+    :param workflow_metadata: Metadata of the workflow to process
+    :return:
+    """
+
+    shards = workflow_metadata["calls"][task]
+
+    shard_status_count = get_shard_status_count(shards)
+
+    shards_done = shard_status_count["Done"] if "Done" in shard_status_count else 0
```

Comment:
This bit has changed since last time but if the suggestion still applies then `get_shard_satus_count` can be edited so that it will return two possible types of results depending on the input. For example, if only shards are given then it returns all the shard's status counts, but if the status is specified then it returns the count for that specific status